### PR TITLE
[lldb][AArch64] Do not crash if NT_ARM_TLS is missing

### PR DESCRIPTION
**Pull request by igorkudrin**
[D156118](https://reviews.llvm.org/D156118) states that this note is always present, but it is better to check it explicitly, as otherwise `lldb` may crash when trying to read registers.

### CODE REVIEW

File: lldb/source/Plugins/Process/Utility/RegisterInfoPOSIX_arm64.cpp [hunk 1]
```diff
@@ -254,9 +254,10 @@ RegisterInfoPOSIX_arm64::RegisterInfoPOSIX_arm64(
       if (m_opt_regsets.AllSet(eRegsetMaskMTE))
         AddRegSetMTE();
 
-      // The TLS set always contains tpidr but only has tpidr2 when SME is
-      // present.
-      AddRegSetTLS(m_opt_regsets.AllSet(eRegsetMaskSSVE));
+      if (m_opt_regsets.AllSet(eRegsetMaskTLS))
```
Comment (DavidSpickett) [inline lldb/source/Plugins/Process/Utility/RegisterInfoPOSIX_arm64.cpp:257]: Add braces here {} as the body is now >1 line.

File: lldb/test/Shell/Process/elf-core/aarch64-no-NT_ARM_TLS.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# REQUIRES: aarch64
```
Comment (DavidSpickett) [inline lldb/test/Shell/Process/elf-core/aarch64-no-NT_ARM_TLS.yaml:1]: Is this requiring the host to be aarch64 or that the aarch64 llvm target is enabled, we should only need the latter for this.

**Comment by DavidSpickett** [comment]
And "don't crash on invalid input" is a good enough reason to fix this regardless of how the core file was generated, even if the core file is corrupted.

File: lldb/source/Plugins/Process/Utility/RegisterInfoPOSIX_arm64.cpp (hunk 1 above)
Comment (igorkudrin) [reply lldb/source/Plugins/Process/Utility/RegisterInfoPOSIX_arm64.cpp]: Done, thanks

File: lldb/test/Shell/Process/elf-core/aarch64-no-NT_ARM_TLS.yaml (hunk 2 above)
Comment (igorkudrin) [reply lldb/test/Shell/Process/elf-core/aarch64-no-NT_ARM_TLS.yaml]: It seems that the test does not need either to work, so I removed the requirement.

**Review by DavidSpickett — APPROVED** [review]
LGTM, thanks!

**Pull request merged**